ReactAndroid/src/main/java/com/facebook/react/modules/fresco/NetworkFetcher.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ * <p/>
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.fresco;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executor;
+
+import android.net.Uri;
+import android.os.Looper;
+import android.os.SystemClock;
+
+import com.facebook.common.logging.FLog;
+import com.facebook.imagepipeline.backends.okhttp3.OkHttpNetworkFetcher;
+import com.facebook.imagepipeline.producers.BaseProducerContextCallbacks;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.ReadableMapKeySetIterator;
+import okhttp3.CacheControl;
+import okhttp3.Callback;
+import okhttp3.Call;
+import okhttp3.Headers;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+class NetworkFetcher extends OkHttpNetworkFetcher {
+
+  private static final String TAG = "NetworkFetcher";
+
+  private final OkHttpClient mOkHttpClient;
+  private final Executor mCancellationExecutor;
+
+  /**
+   * @param okHttpClient client to use
+   */
+  public NetworkFetcher(OkHttpClient okHttpClient) {
+    super(okHttpClient);
+    mOkHttpClient = okHttpClient;
+    mCancellationExecutor = okHttpClient.dispatcher().executorService();
+  }
+
+  private Map<String, String> getHeaders(ReadableMap readableMap) {
+    if (readableMap == null) {
+        return null;
+    }
+    ReadableMapKeySetIterator iterator = readableMap.keySetIterator();
+    Map<String, String> map = new HashMap<>();
+    while (iterator.hasNextKey()) {
+      String key = iterator.nextKey();
+      String value = readableMap.getString(key);
+      map.put(key, value);
+    }
+    return map;
+  }
+
+  @Override
+  public void fetch(final OkHttpNetworkFetchState fetchState, final Callback callback) {
```

Comment:
Is all this code basically just the fetch method from OkHttpNetworkFetcher just so we can inject headers?  Or do you do something fundamentally different?